### PR TITLE
[CUMULUS-1580] Use async-operations to kick off bulk operations

### PR DESCRIPTION
**Pull request by ifestus**
**Summary:** Use async operations to kick off bulk operations.

Addresses [CUMULUS-1580: Use async-operations for kicking off bulk operations](https://bugs.earthdata.nasa.gov/browse/CUMULUS-1580)

## Changes

* Added a new `PUT` endpoint to the granules backend API to handle bulk-granule requests.
* Use built-in Cumulus async-operation functionality to support the aforementioned API.
* Add async-operation Id attribute to executions in DynamoDB (and `cumulus_meta` of those step functions) for future tracking.

## PR Checklist

- [ ] Update CHANGELOG
- [x] Unit tests
- [x] Adhoc testing
- [ ] Integration tests

### CODE REVIEW

File: example/lambdas.yml [hunk 1]
```diff
@@ -136,7 +136,6 @@ SfSnsReport:
 SyncGranule:
   handler: index.handler
   timeout: 300
-  logToElasticSearch: true
```
Comment (laurenfrederick) [inline example/lambdas.yml]: @ifestus Is this on purpose? What is the reason for this change?

Comment (ifestus) [reply example/lambdas.yml]: Good catch. I removed it so I could set `logToSharedDestination` to true, since we can't set more than one log destination for a log group. I'll add that line back :+1:
